commons/com.b2international.index/src/com/b2international/index/revision/StagingArea.java

```diff
@@ -673,43 +673,41 @@ void merge(RevisionBranchRef fromRef, RevisionBranchRef toRef, boolean squash, R
+				final Map<String, Map<String, RevisionCompareDetail>> targetPropertyChangesByObject = indexPropertyChangesByObject(toChangeDetails);
 				for (String changedInSourceAndTargetId : changedInSourceAndTargetIds) {
-					Map<String, RevisionCompareDetail> sourcePropertyChanges = fromChangeDetails.stream()
-							.filter(detail -> detail.getObject().id().equals(changedInSourceAndTargetId))
-							.filter(detail -> !detail.isComponentChange())
-							.collect(Collectors.toMap(RevisionCompareDetail::getProperty, d -> d));
-					Map<String, RevisionCompareDetail> targetPropertyChanges = toChangeDetails.stream()
-							.filter(detail -> detail.getObject().id().equals(changedInSourceAndTargetId))
-							.filter(detail -> !detail.isComponentChange())
-							.collect(Collectors.toMap(RevisionCompareDetail::getProperty, d -> d));
+					final Map<String, RevisionCompareDetail> sourcePropertyChanges = sourcePropertyChangesByObject.get(changedInSourceAndTargetId);
+					final Map<String, RevisionCompareDetail> targetPropertyChanges = targetPropertyChangesByObject.get(changedInSourceAndTargetId);
 					
-					for (Entry<String, RevisionCompareDetail> sourceChange : Iterables.consumingIterable(sourcePropertyChanges.entrySet())) {
-						final RevisionPropertyDiff sourceChangeDiff = new RevisionPropertyDiff(sourceChange.getValue().getProperty(), sourceChange.getValue().getFromValue(), sourceChange.getValue().getValue());
-						final RevisionCompareDetail targetPropertyChange = targetPropertyChanges.remove(sourceChange.getKey());
-						if (targetPropertyChange == null) {
-							// this property did not change in target, just apply directly on the target object via
-							if (!propertyUpdatesToApply.containsKey(type)) {
-								propertyUpdatesToApply.put(type, HashMultimap.create());
-							}
-							propertyUpdatesToApply.get(type).put(changedInSourceAndTargetId, sourceChangeDiff);
-							fromChangeSet.removeChanged(type, changedInSourceAndTargetId);
-						} else {
-							RevisionPropertyDiff targetChangeDiff = new RevisionPropertyDiff(targetPropertyChange.getProperty(), targetPropertyChange.getFromValue(), targetPropertyChange.getValue());
-							// changed on both sides, ask conflict processor to resolve the issue or raise conflict error
-							RevisionPropertyDiff resolution = conflictProcessor.handleChangedInSourceAndTarget(
-								changedInSourceAndTargetId, 
-								sourceChangeDiff,
-								targetChangeDiff
-							);
-							if (resolution == null) {
-								conflicts.add(new ChangedInSourceAndTargetConflict(sourceChange.getValue().getObject(), sourceChangeDiff.convert(conflictProcessor), targetChangeDiff.convert(conflictProcessor)));
-							} else {
+					if (sourcePropertyChanges != null) {
+						for (Entry<String, RevisionCompareDetail> sourceChange : Iterables.consumingIterable(sourcePropertyChanges.entrySet())) {
```

Comment:
It no longer seems to be necessary to consume Map entries here 🤔 We're not doing anything with any leftover entries in `sourcePropertyChanges` (or indirectly via `sourcePropertyChangesByObject`).